plugins/inputs/gnmi/gnmi.go

```diff
@@ -296,6 +312,22 @@ func (c *GNMI) handleSubscribeResponseUpdate(address string, response *gnmi.Subs
 			}
 		}
 
+		// Update tag lookups and discard rest of update
+		if lu, ok := c.lookup[name]; ok {
+			updateLookups(lu[tags["source"]], tags, fields)
+			continue
+		}
+
+		// Apply lookups if present
+		for k, v := range c.lookup {
+			if t, ok := v[tags["source"]][tags["name"]]; ok {
+				for name, val := range t {
+					tagName := fmt.Sprintf("%s_%s", k, name)
+					tags[tagName] = val.(string)
+				}
+			}
+		}
```

Comment:
Similar to the previous comment of Steven, I'd like to see this lookup simplified. So how about using a `map[string]map[string]map[string]string` for `c.lookup`. This would save one `map`... The code here then looks like the following
```suggestion
		subscriptionKey := tags["source"] + "/" + tags["name"]
		if _, ok := c.lookup[name]; ok {
			// We are subscribed to this, so add the fields to the lookup-table
			if _, ok := c.lookup[name][subscriptionKey]; !ok {
				c.lookup[name][subscriptionKey] = make(map[string]string)
			}
			for k, v := range fields {
				c.lookup[name][subscriptionKey][path.Base(k)] = v
			}
			// Do not process the data further as we only subscribed here to update the lookup-table
			continue
		}

		// Apply lookups if present
		for subscriptionName, values := range c.lookup {
			annotations, ok := values[subscriptionKey]
			if !ok {
				continue
			}
			for k, v := range annotations {
				tags[subscriptionName + "_" + k] = v
			}
		}
```

Reply:
Thanks for the suggestion!  I did collapse the two values into a single subscription key, but kept the bottom level of the map as an interface, in case there's a use case where someone wants to use a non-string metric as a tag.